package/kedro_viz/models/experiment_tracking.py

```diff
@@ -78,11 +85,18 @@ def load_tracking_data(self, run_id: str):
             # tracking datasets do not have load methods defined yet but would be the
             # same as json loader.
             # pylint: disable=import-outside-toplevel
-            from kedro.extras.datasets import json, tracking
+            from kedro.extras.datasets import json, tracking, plotly, matplotlib
 
             tracking.JSONDataSet._load = json.JSONDataSet._load  # type: ignore
             tracking.MetricsDataSet._load = json.JSONDataSet._load  # type: ignore
-            self.runs[run_id] = self.dataset.load()
+            plotly.JSONDataSet._load =  json.JSONDataSet._load  # type: ignore
+            plotly.PlotlyDataSet._load = json.JSONDataSet._load 
+            matplotlib.MatplotlibWriter._load = matplotlib_writer_load
+
+            if isinstance(self.dataset, (PlotlyJSONDataSet, PlotlyDataSet, MatplotlibWriter)):
+                self.runs[run_id] = {get_filename(self.dataset): self.dataset.load()}
```

Comment:
```suggestion
                self.runs[run_id] = {self.dataset._filepath.name: self.dataset.load()}
```
Unless there's some reason we need to use `get_filename` as you wrote it?